internal/cluster/distribution/eks/eksadapter/node_pool_manager.go

```diff
@@ -93,5 +120,141 @@ func (n nodePoolManager) UpdateNodePool(
 
 // TODO: this is temporary
 func generateNodePoolStackName(clusterName string, poolName string) string {
-	return "pipeline-eks-nodepool-" + clusterName + "-" + poolName
+	return nodePoolStackNamePrefix + clusterName + "-" + poolName
+}
+
+// ListNodePools lists node pools from a cluster.
+func (n nodePoolManager) ListNodePools(ctx context.Context, c cluster.Cluster, nodePoolNames []string) ([]eks.NodePool, error) {
+	clusterClient, err := n.dynamicClientFactory.FromSecret(ctx, c.ConfigSecretID.String())
+	if err != nil {
+		return nil, errors.WrapWithDetails(err, "creating dynamic Kubernetes client factory failed", "cluster", c)
+	}
+
+	manager := npls.NewManager(clusterClient, n.namespace)
+	labelSets, err := manager.GetAll()
+	if err != nil {
+		return nil, errors.WrapWithDetails(err, "retrieving node pool label sets failed",
+			"cluster", c,
+			"namespace", n.namespace,
+		)
+	}
+
+	awsClient, err := n.awsFactory.New(c.OrganizationID, c.SecretID.ResourceID, c.Location)
+	if err != nil {
+		return nil, errors.WrapWithDetails(err, "creating aws factory failed", "cluster", c)
+	}
+
+	cfClient := n.cloudFormationFactory.New(awsClient, nil)
+	describeStacksInput := cloudformation.DescribeStacksInput{}
+	nodePools := make([]eks.NodePool, len(nodePoolNames))
```

Comment:
I think it's nicer to define the slice with 0 len and the appropriate capacity and use `append`, that way you don't have to track the index. Since the capacity will never run out, appending has no performance penalty of reslicing.

Reply:
Fixed in https://github.com/banzaicloud/pipeline/compare/2ad4cafb067c442a1cfbf1b9cd4071e298b269c8..f640984ef5dee7ae5770cc55c9ccc9741c0e760b.